src/ngraph/runtime/gpu/gpu_cuda_kernel_builder.cpp

```diff
@@ -483,9 +505,25 @@ void runtime::gpu::CudaKernelBuilder::get_reduce_to_nd_op(
     runtime::gpu::GPUKernelArgs& args,
     const std::vector<std::string>& data_types,
     const std::string& reduce_op,
-    size_t out_rank,
+    size_t non_reduce_rank,
     size_t reduce_rank)
 {
+    bool stable_sum =
+        ((reduce_op == "add") && (data_types[1] == "float" || data_types[1] == "double"));
+    auto stable_sum_lambda = [&]() {
```

Comment:
Nice. Will this stable_sum snippet be used elsewhere (e.g. Softmax)? If so, maybe make it a helper function instead of an in context lambda.

Reply:
nit: Maybe the codegen itself could be in the helper?